src/getMigrateDashboards.ts

```diff
@@ -54,13 +60,24 @@ export const getMigrateDashboards =
     const { content, structure } =
       contentServer.children?.ui.children?.dashboards.children ?? {};
 
-    if (!content?.children || !structure?.children) {
+    if (!content?.children || !structure?.children || !originalContent) {
       return;
     }
 
     const filesAncestry = _getFilesAncestry(structure);
 
     for (const fileId in content.children) {
+      if (
+        errorReport.dashboards?.[fileId] &&
+        !isDashboardErrorReport(errorReport.dashboards?.[fileId]) &&
+        behaviorOnError !== "keep-going"
+      ) {
+        // The migration of this dashboard failed at a previous step.
+        // It is is not a partial failure (i.e. an error on the migration of one of its widgets).
+        // The behavior on error is not to keep going, hence the migration of this dashboard should not go further.
```

Comment:
I would remove these lines as I think the rest speaks for itself (or will with the renaming suggestion above).
```suggestion
```